pyvrp/PenaltyManager.py

```diff
@@ -124,6 +124,23 @@ def __init__(
         self._time_feas: List[bool] = []  # track recent time feasibility
         self._capacity_penalty = params.init_capacity_penalty
         self._tw_penalty = params.init_tw_penalty
+        self._cost_evaluator = CostEvaluator(
+            self._capacity_penalty, self._tw_penalty
+        )
+        self._booster_cost_evaluator = CostEvaluator(
+            self._capacity_penalty * self._params.repair_booster,
+            self._tw_penalty * self._params.repair_booster,
+        )
+
+    def _update_cost_evaluators(self):
+        # Updates the cost evaluators given new penalty values
+        self._cost_evaluator = CostEvaluator(
+            self._capacity_penalty, self._tw_penalty
+        )
+        self._booster_cost_evaluator = CostEvaluator(
+            self._capacity_penalty * self._params.repair_booster,
+            self._tw_penalty * self._params.repair_booster,
+        )
```

Comment:
Do these need to be cached? They're simple objects to create and destroy, is that really costing us?

Reply:
See my other comment, are you ok with keeping it or do you really think it should be changed?

BTW independently of the question whether it's needed I agree this code is not very pretty as it duplicates the initialization code. However I could not see (in general) the best solution? Do you have a suggestion?
* Not defining the properties in the constructor is bad practice.
* Initializing to `None` gets `mypy` complaining. Typing as optional seems overkill but maybe disable the check?
* We do not (yet) have a default constructor for `CostEvaluator` and using this would also not be pretty.

Reply:
I don't see a clean way of avoiding this, other than calling `self._update_cost_evaluators()` from the initialiser. Let's leave it as-is.

I ran some experiments with instantiating `CostEvaluator` a million times or getting it from a variable. The variable is some 6x faster. I doubt it matters a lot, but we can keep it in.

Reply:
Thanks for the experiments! I think its good that we have the discussion and aim to avoid pre-mature optimization in general but here it feels appropriate, now we won't have to worry it could become a bottleneck either. :)